src/sentry/static/sentry/app/components/confirm.jsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Modal from 'react-bootstrap/lib/Modal';
+
+import Button from './buttons/button';
+
+class Confirm extends React.PureComponent {
+  static propTypes = {
+    disabled: PropTypes.bool,
+    message: PropTypes.string.isRequired,
+    onConfirm: PropTypes.func.isRequired,
+    priority: PropTypes.oneOf(['primary', 'danger']).isRequired,
+    confirmText: PropTypes.string.isRequired,
+    cancelText: PropTypes.string.isRequired
+  };
+
+  static defaultProps = {
+    priority: 'primary',
+    cancelText: 'Cancel',
+    confirmText: 'Confirm'
+  };
+
+  constructor(...args) {
+    super(...args);
+    this.state = {
+      isModalOpen: false
+    };
+  }
+
+  handleConfirm = () => {
+    this.setState({
+      isModalOpen: false
+    });
+
+    this.props.onConfirm();
+  };
+
+  handleToggle = () => {
+    if (this.props.disabled) {
+      return;
+    }
+    this.setState({
+      isModalOpen: !this.state.isModalOpen
```

Comment:
this could be another place that double triggering could occur? maybe this is an imagined failure case

Reply:
Currently, Bootstrap's Modal seems to be sure that this doesn't happen (the second click causes modal to hide). Just in case I've added a test case in case we change Modal libs.